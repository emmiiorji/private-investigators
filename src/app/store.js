import { configureStore } from '@reduxjs/toolkit';
import investigatorsReducer from './redux/investigator/investigatorSlice';

const store = configureStore({
  reducer: {
    investigators: investigatorsReducer,
  },
});

export default store;
