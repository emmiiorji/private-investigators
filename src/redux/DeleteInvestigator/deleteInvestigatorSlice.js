import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

export const deleteInvestigator = createAsyncThunk('investigators/delete', async (investigatorId) => {
  await axios.delete(`${process.env.REACT_APP_SERVER_BASE_URL}/v1/investigators/${investigatorId}`);
  return investigatorId;
});

const initialState = {
  value: [],
  status: 'idle',
};

const deleteInvestigatorSlice = createSlice({
  name: 'investigators/delete',
  initialState,
  reducers: {
    loadInvestigator: (state, action) => {
      const newState = { ...state, value: [...state.value, action.payload] };
      return newState;
    },
  },

  extraReducers: (builder) => {
    builder
      .addCase(deleteInvestigator.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(deleteInvestigator.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.error.message;
      })
      .addCase(deleteInvestigator.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.value = state.value.filter(
          (investigator) => investigator.id !== parseInt(action.payload, 10),
        );
      });
  },
});

export default deleteInvestigatorSlice.reducer;
