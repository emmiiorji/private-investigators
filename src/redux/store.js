import { configureStore } from '@reduxjs/toolkit';
// import userReducer from './CurrentUser/CurrentUserSlice';
import investigatorsReducer from './investigator/investigatorSlice';
import appointmentsReducer from './Appointments/appointmentsSlice';
import userReducer from './CurrentUser/CurrentUserSlice';

const store = configureStore({
  reducer: {
    investigators: investigatorsReducer,
    appointments: appointmentsReducer,
    userReducer,
  },
});

export default store;
