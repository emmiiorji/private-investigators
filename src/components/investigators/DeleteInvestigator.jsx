import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { deleteInvestigator } from '../../redux/DeleteInvestigator/deleteInvestigatorSlice';
import { Button, Card } from 'react-bootstrap';

const DeleteInvestigator = () => {
  const investigators = useSelector((state) => state.investigators);

  const dispatch = useDispatch();

  return (
    <div>
      <h1>Delete an Investigator</h1>
      <div>
        {investigators.value.map((investigator) => (
          <Card key={investigator.id} className="d-flex flex-row justify-content-between m-4 p-4">
            <img src={`${investigator.photo}`} alt="photo" style={{maxHeight: "2rem"}} />
            {investigator.name}
            <Button
              type="button"
              id={investigator.id}
              onClick={handleDelete}
            >
              Delete
            </Button>
          </Card>
        ))}
      </div>
    </div>
  );
};

export default DeleteInvestigator;