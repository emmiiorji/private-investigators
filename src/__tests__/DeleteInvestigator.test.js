import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DeleteInvestigator from '../components/investigators/DeleteInvestigator';

describe('RootLayout', () => {
  let deleteInvestigator;
  beforeAll(async () => {
    deleteInvestigator = renderer
      .create(
        <Provider store={store}>
          <Router>
            <DeleteInvestigator />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('It should render correctly', () => {
    expect(deleteInvestigator).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(deleteInvestigator).toMatchSnapshot();
  });

  it('It displays correct message when there is no any investigator', () => {
    const errMessage = waitFor(() => screen.getByText(/No investigator is available!/i));
    waitFor(() => expect(errMessage).toBeInTheDocument());
  });
});
