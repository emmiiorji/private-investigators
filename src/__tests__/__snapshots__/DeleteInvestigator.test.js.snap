// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout It should match its snapshot 1`] = `
<h1>
  Delete Form
</h1>
`;
